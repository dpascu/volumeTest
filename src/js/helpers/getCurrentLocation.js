const DEF_POS = { lat: 51.5074, lng: 0.1278 };

/**
 * Fetches current coordinates, or defaults one if it doesn't work
 * @returns {Object} {lat, lng}
 */
const getCurrentLocation = () =>
  new Promise((res) => {
    if (!navigator.geolocation || !navigator.geolocation.getCurrentPosition) res(DEF_POS);
    navigator.geolocation.getCurrentPosition((pos) => {
      const { latitude: lat, longitude: lng } = pos.coords;
      res({ lat, lng });
    }, () => {
      res(DEF_POS);
    });
  });

export default getCurrentLocation;
