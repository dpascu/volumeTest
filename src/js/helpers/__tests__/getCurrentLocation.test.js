/* global describe it expect */

import getCurrentLocation from '../getCurrentLocation';

describe('getCurrentLocation', () => {
  it('should fetch data', async () => {
    const weather = await getCurrentLocation();
    expect(weather).toMatchSnapshot();
  });
});
