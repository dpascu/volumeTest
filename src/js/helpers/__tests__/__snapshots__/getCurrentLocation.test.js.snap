// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCurrentLocation should fetch data 1`] = `
Object {
  "lat": 51.5074,
  "lng": 0.1278,
}
`;
